dipy/nn/torch/evac.py

```diff
@@ -0,0 +1,551 @@
+            )
+            self.layer_list2.append(ReLU())
+
+        self.channel_sum = ChannelSum()
+        self.add = Add()
+
+    def forward(self, x, passed):
+        for layer in self.layer_list:
+            x = layer(x)
+
+        x = self.channel_sum(x)
+        fwd = self.add(x, passed)
+        x = fwd
+
+        for layer in self.layer_list2:
+            x = layer(x)
+
+        return fwd, x
+
+
+class Model(Module):
+    def __init__(self, model_scale=16):
+        super(Model, self).__init__()
+
+        # Block structure
+        self.block1 = Block(
+            1, model_scale, kernel_size=5, strides=1, padding=2, drop_r=0.2, n_layers=1
+        )
+        self.block2 = Block(
+            2,
+            model_scale * 2,
+            kernel_size=5,
+            strides=1,
+            padding=2,
+            drop_r=0.5,
+            n_layers=2,
+        )
+        self.block3 = Block(
+            2,
+            model_scale * 4,
+            kernel_size=5,
+            strides=1,
+            padding=2,
+            drop_r=0.5,
+            n_layers=3,
+        )
+        self.block4 = Block(
+            2,
+            model_scale * 8,
+            kernel_size=5,
+            strides=1,
+            padding=2,
+            drop_r=0.5,
+            n_layers=3,
+        )
+        self.block5 = Block(
+            2,
+            model_scale * 16,
+            kernel_size=5,
+            strides=1,
+            padding=2,
+            drop_r=0.5,
+            n_layers=3,
+            passed_channel=2,
+            layer_type="up",
+        )
+
+        # Upsample/decoder blocks
+        self.up_block1 = Block(
+            3,
+            model_scale * 8,
+            kernel_size=5,
+            strides=1,
+            padding=2,
+            drop_r=0.5,
+            n_layers=3,
+            passed_channel=1,
+            layer_type="up",
+        )
+        self.up_block2 = Block(
+            3,
+            model_scale * 4,
+            kernel_size=5,
+            strides=1,
+            padding=2,
+            drop_r=0.5,
+            n_layers=3,
+            passed_channel=1,
+            layer_type="up",
+        )
+        self.up_block3 = Block(
+            3,
+            model_scale * 2,
+            kernel_size=5,
+            strides=1,
+            padding=2,
+            drop_r=0.5,
+            n_layers=2,
+            passed_channel=1,
+            layer_type="up",
+        )
+        self.up_block4 = Block(
+            2,
+            model_scale,
+            kernel_size=5,
+            strides=1,
+            padding=2,
+            drop_r=0.5,
+            n_layers=1,
+            passed_channel=1,
+            layer_type="none",
+        )
+
+        self.conv_pred = Conv3d(1, 2, 1, padding=0)
+        self.softmax = Softmax(dim=1)
+
+    def forward(self, inputs, raw_input_2, raw_input_3, raw_input_4, raw_input_5):
+        fwd1, x = self.block1(inputs, inputs)
```

Comment:
same comment as above concerning `x`

Reply:
I believe in that case it will be holding in memory unnecessary intermediate results. I could change it if needed though.